FoxySigner

ndvo opened the pull request:
Implements a utility to create HMAC signatures for both URL and fields.

It creates a hmacSign attribute in FoxyApi which provides an instance of FoxySigner.

ndvo committed:
build: add crypto definition files

ndvo committed:
chore(.gitignore): add vim, ctags and global files to gitignore

ndvo committed:
refactor: expose AuthInit type to be used in tests

ndvo committed:
chore(tests): create a mock config for creating auth obj

ndvo committed:
feat: add a signer object to FoxyApi

Add a signer object to the FoxyApi. So far, it only signs simple messages with the clientSecret.

ndvo committed:
test(signer): test signing input and query fields

ndvo committed:
feat(signer): hmac signatures for query and input fields

ndvo committed:
build: add jsdom as a dependency

ndvo committed:
feat(signer): [partial] methods to sign elements

Signs elements, pages and files. This is partial commit. The funcionality is not yet done.

BREAKING CHANGE: The funcionality is only partially implemented.

ndvo committed:
refactor(signer): moves codes dict types folder

ndvo committed:
feat(signer): signs a query string

ndvo committed:
docs(signer): add signing instructions

ndvo committed:
feat(signer): sign html files

ndvo committed:
refactor: clean up the api interface

ndvo committed:
feat(signer): sign files asynchronously

ndvo committed:
refactor(signer): rename queryString to url

ndvo committed:
test(signer): test editable field

ndvo committed:
test(signer): test textarea

ndvo committed:
test(signer): test radio button

ndvo committed:
fix(signer): fix error fetching code of products when signing forms

ndvo committed:
test(signer): add missing tests

ndvo committed:
test(signer): test if it signs forms without a code value

ndvo committed:
fix(signer): queryStringArg must always have a value

ndvo committed:
docs(signer): adjust the documentation to the new names

ndvo committed:
refactor(signer): moves signer to utils

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Foxy/foxy-node-api?pullRequest=5) <br/>All committers have signed the CLA.

brettflorio committed:
docs(signer): Minor improvements to hmac docs.

pheekus committed:
build: add jsdoc plugin for eslint

pheekus committed:
docs(signer): make improvements to the jsdoc comments

pheekus committed:
refactor(signer): remove unused code

pheekus committed:
refactor(signer): use more generic type for fragment signing

The `FoxySigner#fragment()` method signs a fragment of the document, not `DocumentFragment` specifically, so using a more generic type would make more sense in this case.

pheekus committed:
fix(signer): remove private keyword for public methods

pheekus committed:
build(eslint): enable naming convention rules

pheekus committed:
style(signer): follow naming convention

pheekus committed:
refactor: remove Signer interface

While this could be considered a good practice or even a requirement in other languages, it's a bit too verbose for the number of benefits it brings to a TypeScript project.

pheekus committed:
test(signer): remove unused code

ndvo committed:
refactor(signer): make methods private, following naming convention

ndvo committed:
refactor(signer): check for propper URL before signing it

ndvo committed:
docs(signer): adapt private methods to the documentation convention

ndvo committed:
refactor(signer): remove unused code

ndvo committed:
refactor(signer): set getCodeFromURL method private

ndvo committed:
test(signer): remove direct tests of private methods

ndvo committed:
test(signer): test for when code is not the first argument

ndvo committed:
test(signer): replace direct tests of private methods

pheekus committed:
test(signer): remove unused code

pheekus reviewed (approved):
Awesome job, @ndvo, and thanks for making those changes to the tests! @brettflorio feel free to merge if everything's ok on your end.
